java/core/src/main/java/co/worklytics/psoxy/gateway/impl/GoogleCloudPlatformServiceAccountKeyAuthStrategy.java

```diff
@@ -122,7 +123,9 @@ GoogleCredentials buildImpersonatedCredentials(@NonNull String accountToImperson
 
         //even though GoogleCredentials implements `createDelegated`, it's a no-op if the
         // credential type doesn't support it.
+        // similarly, createScoped() is no-op if not supported by GoogleCredentials implementation
         return baseCredentials
+            .toBuilder().build() // force a copy of base, which we'll then mutate to be delegated, scoped
```

Comment:
so actually I'm wrong, it's OK in the cases that matter: 
![psoxy____code_psoxy__–_ServiceAccountCredentials_java__Maven__com_google_auth_google-auth-library-oauth2-http_1_18_0_](https://github.com/Worklytics/psoxy/assets/102831/7fbc840c-14fb-42bc-ae8c-f553adb86b4b)

Reply:
reverting; i had done this thinking it might be the source of auth problem; but it's OK